src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -951,103 +973,98 @@ private void consume(final String queue) {
                                 }
-                                redisClient.lpop(key1, jsonAnswer -> {
+                                redisClient.lpop(queueKey, jsonAnswer -> {
                                     if (jsonAnswer.failed()) {
-                                        log.error("Failed to pop from queue '{}'", queue, jsonAnswer.cause());
+                                        log.error("Failed to pop from queue '{}'", queueName, jsonAnswer.cause());
                                         // We should return here. See: "https://softwareengineering.stackexchange.com/a/190535"
                                     }
-                                    log.debug("RedisQues Message removed, queue " + queue + " is ready again");
-                                    myQueues.put(queue, QueueState.READY);
-                                    // Notify that we are stopped in
-                                    // case it
-                                    // was the last active consumer
+                                    log.debug("RedisQues Message removed, queue " + queueName + " is ready again");
+                                    myQueues.put(queueName, QueueState.READY);
+                                    // Notify that we are stopped in case it was the last active consumer
                                     if (stoppedHandler != null) {
                                         unregisterConsumers(false);
                                         if (myQueues.isEmpty()) {
                                             stoppedHandler.handle(null);
                                         }
                                     }
                                     // Issue notification to consume next message if any
-                                    String key2 = getQueuesPrefix() + queue;
                                     if (log.isTraceEnabled()) {
-                                        log.trace("RedisQues read queue: " + key);
+                                        log.trace("RedisQues read queue: " + queueKey);
                                     }
-                                    redisClient.llen(key2, answer1 -> {
+                                    redisClient.llen(queueKey, answer1 -> {
                                         if (answer1.succeeded() && answer1.result() != null && answer1.result() > 0) {
-                                            notifyConsumer(queue);
+                                            notifyConsumer(queueName);
                                         }
                                     });
                                 });
                             } else {
                                 // Failed. Message will be kept in queue and retried later
-                                log.debug("RedisQues Processing failed for queue " + queue);
+                                log.debug("RedisQues Processing failed for queue " + queueName);
```

Comment:
What about
```java
if (log.isDebugEnabled()) {
	//.....
}
```
?

Reply:
This is a rare case (processor failed) - but will do. Thx